From ff6e8675b61261d4e85e7fb1a4b461dec2d13abe Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Thu, 9 Nov 2023 15:46:51 +0100
Subject: [PATCH 0678/1141] dts: arm64: st: OP-TEE async notif on PPI 15 for
 stm32mp25

Define GIC PPI 15 (aka GIC interrupt line 31) for OP-TEE asynchronous
notification.

Change-Id: Id5216132ae78fb937661be13dd03e5064f7ea04b
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/338812
Reviewed-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 arch/arm64/boot/dts/st/stm32mp251.dtsi | 10 +++++++---
 arch/arm64/boot/dts/st/stm32mp253.dtsi |  4 ++++
 2 files changed, 11 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/st/stm32mp251.dtsi b/arch/arm64/boot/dts/st/stm32mp251.dtsi
index 16f397f6b194..f9618bf10f7e 100644
--- a/arch/arm64/boot/dts/st/stm32mp251.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp251.dtsi
@@ -91,9 +91,13 @@ clk_rcbsec: clk-rcbsec {
 	};
 
 	firmware {
-		optee {
+		optee: optee {
 			compatible = "linaro,optee-tz";
 			method = "smc";
+			interrupt-parent = <&intc>;
+			interrupts = <GIC_PPI 15 (GIC_CPU_MASK_SIMPLE(1) | IRQ_TYPE_LEVEL_LOW)>;
+			interrupt-controller;
+			#interrupt-cells = <1>;
 		};
 
 		scmi: scmi {
@@ -696,8 +700,8 @@ lptimer1: timer@40090000 {
 				#address-cells = <1>;
 				#size-cells = <0>;
 				feature-domains = <&rifsc STM32MP25_RIFSC_LPTIM1_ID>;
-				wakeup-source;
 				power-domains = <&RET_PD>;
+				wakeup-source;
 				status = "disabled";
 
 				counter {
@@ -732,8 +736,8 @@ lptimer2: timer@400a0000 {
 				#address-cells = <1>;
 				#size-cells = <0>;
 				feature-domains = <&rifsc STM32MP25_RIFSC_LPTIM2_ID>;
-				wakeup-source;
 				power-domains = <&RET_PD>;
+				wakeup-source;
 				status = "disabled";
 
 				counter {
diff --git a/arch/arm64/boot/dts/st/stm32mp253.dtsi b/arch/arm64/boot/dts/st/stm32mp253.dtsi
index dfa0177c9138..d205f3712eeb 100644
--- a/arch/arm64/boot/dts/st/stm32mp253.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp253.dtsi
@@ -141,3 +141,7 @@ &ltdc {
 	clocks = <&rcc CK_BUS_LTDC>, <&rcc CK_KER_LTDC>, <&syscfg 0>;
 	clock-names = "bus", "ref", "lcd";
 };
+
+&optee {
+	interrupts = <GIC_PPI 15 (GIC_CPU_MASK_SIMPLE(2) | IRQ_TYPE_LEVEL_LOW)>;
+};
-- 
2.39.2

